lib/common/acl.c

```diff
@@ -585,9 +585,23 @@ pcmk__apply_creation_acl(xmlNode *xml, bool check_top)
                       xml->name, display_id(xml));
 
         } else if (check_top) {
-            crm_trace("ACLs disallow creation of <%s> with "
-                      PCMK_XA_ID "=\"%s\"", xml->name, display_id(xml));
-            pcmk_free_xml_subtree(xml);
+            /* is_root=true should be impossible with check_top=true, but check
+             * for sanity
+             */
+            bool is_root = (xml->doc != NULL)
+                           && (xmlDocGetRootElement(xml->doc) == xml);
+
+            crm_trace("ACLs disallow creation of %s<%s> with "
+                      PCMK_XA_ID "=\"%s\"",
+                      (is_root? "root element " : ""), xml->name,
+                      display_id(xml));
+
+            if (is_root) {
+                xmlFreeDoc(xml->doc);
+            } else {
+                xmlUnlinkNode(xml);
```

Comment:
This preserves behavior, but it triggers a thought. I haven't traced this, but at a glance, it seems that we could:
1. create a new XML node
2. mark the parent dirty
3. unlink/free the newly created node due to ACL denial
4. the parent is still dirty even though nothing has changed anymore

Not sure if that has any downstream effects (including but not limited to unexpected log messages/diffs).